packages/babel-parser/src/parser/expression.js

```diff
@@ -1208,12 +1232,49 @@ export default class ExpressionParser extends LValParser {
           }
         }
       }
+
       // fall through
       default:
         throw this.unexpected();
     }
   }
 
+  // https://github.com/js-choi/proposal-hack-pipes
+  parseTopicReference() {
+    const pipeProposal = this.getPluginOption("pipelineOperator", "proposal");
+
+    if (pipeProposal) {
+      const node = this.startNode();
+
+      let nodeType;
+      if (pipeProposal === "hack") {
+        nodeType = "TopicReference";
+      } else if (pipeProposal === "smart") {
+        nodeType = "PipelinePrimaryTopicReference";
+      } else {
+        throw this.raise(node.start, Errors.PipeTopicRequiresHackPipes);
+      }
+
+      this.next();
+
+        if (!this.topicReferenceIsAllowedInCurrentContext()) {
+          switch (pipeProposal) {
+            case "hack":
+              this.raise(node.start, Errors.PipeTopicUnbound);
+              break;
+            case "smart":
+              this.raise(node.start, Errors.PrimaryTopicNotAllowed);
+              break;
+          }
+        }
+      }
```

Comment:
I think there should be one `}` less?

Reply:
Yes, this was my mistake. My commits got mixed up between my two local branches, so I had to rewrite a bit of history, sorry. I force-pushed a corrected version just now.